src/ngraph/runtime/gpu/gpu_emitter.cpp

```diff
@@ -263,23 +265,64 @@ cudnnSetOpTensorDescriptor(opTensorDesc,
                     writer.indent--;
                     writer << "}\n";
                 }
-                else if ((arg0_shape.size() == 2) && (arg1_shape.size() == 2) &&
-                         (dot->get_reduction_axes_count() == 1))
+                // cases that can be treat as matrix multiply
+                else
                 {
-                    // GEMM Call
-                    if (arg0_shape[0] != out[0].get_shape()[0] || // m
-                        arg1_shape[1] != out[0].get_shape()[1] || // n
-                        arg0_shape[1] != arg1_shape[0])           // k
+                    // treat as out[m,n] = arg0[m,k] * arg1[k,n]
```

Comment:
Thanks, it reads better. I've been working through the mathematics of NumPy/Tensorflow tensordot and I realized that op::Dot is less general, is that correct? There isn't any mention of contraction in ngraph op::Dot so I assume so. The matrix multiplication impl. looks good.

Reply:
There was some spec on generalized dot, 
https://github.com/NervanaSystems/ngraph/blob/2246d9ba05ace2368494bb098da0b4964437d76a/src/ngraph/ops/dot.hpp
Let me check with @aprocter

Reply:
The intended semantics is the same as https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.tensordot.html, in the case where `axes` is an "integer_like". In other words, you cannot arbitrarily pair reduction axes in nGraph, unlike NumPy and TF.

Reply:
Thanks @aprocter